legacy/src/main/java/org/opensearch/sql/legacy/cursor/DefaultCursor.java

```diff
@@ -146,58 +142,90 @@ public String generateCursorId() {
                     try {
                       return objectMapper.writeValueAsString(sortFields);
                     } catch (JsonProcessingException e) {
-                      throw new RuntimeException(e);
+                      throw new RuntimeException(
+                          "Failed to parse sort fields from JSON string.", e);
                     }
                   });
       json.put(SORT_FIELDS, sortFieldValue);
+      setSearchRequestString(json, searchSourceBuilder);
     } else {
       json.put(SCROLL_ID, scrollId);
     }
-    return String.format("%s:%s", type.getId(), encodeCursor(json, searchSourceBuilder));
+    return String.format("%s:%s", type.getId(), encodeCursor(json));
+  }
+
+  private void setSearchRequestString(JSONObject cursorJson, SearchSourceBuilder sourceBuilder) {
```

Comment:
nit: we should rather return the searchRequestBase64 from this method and set it to json at caller side. (it is not a good practice to modify parameter object in a method for readability/maintainability) Please address it in later PR.